src/pages/workspace/accounting/AccountingContext.tsx

```diff
@@ -0,0 +1,138 @@
+    shouldDisconnectIntegrationBeforeConnecting?: boolean;
+    integrationToDisconnect?: ConnectionName;
+};
+
+type ActiveIntegrationState = ActiveIntegration & {key: number};
+
+type AccountingContextType = {
+    activeIntegration?: ActiveIntegration;
+    startIntegrationFlow: (activeIntegration: ActiveIntegration) => void;
+
+    /*
+     * This stores refs to integration buttons, so the PopoverMenu can be positioned correctly
+     */
+    integrationRefs: RefObject<Record<string, MutableRefObject<View | null>>>;
+};
+
+const integrationRefsInitialValue = Object.values(CONST.POLICY.CONNECTIONS.NAME).reduce((acc, key) => {
+    acc[key] = {current: null};
+    return acc;
+}, {} as Record<ConnectionName, MutableRefObject<View | null>>);
+
+const defaultAccountingContext = {
+    activeIntegration: undefined,
+    startIntegrationFlow: () => {},
+    integrationRefs: {
+        current: integrationRefsInitialValue,
+    },
+};
+
+const AccountingContext = React.createContext<AccountingContextType>(defaultAccountingContext);
+
+type AccountingContextProviderProps = ChildrenProps & {
+    policy: Policy;
+};
+
+function AccountingContextProvider({children, policy}: AccountingContextProviderProps) {
+    const integrationRefs = useRef<Record<string, MutableRefObject<View | null>>>(defaultAccountingContext.integrationRefs.current);
+    const [activeIntegration, setActiveIntegration] = useState<ActiveIntegrationState>();
+    const {translate} = useLocalize();
+    const policyID = policy.id;
+
+    const startIntegrationFlow = React.useCallback(
+        (newActiveIntegration: ActiveIntegration) => {
+            const accountingIntegrationData = getAccountingIntegrationData(newActiveIntegration.name, policyID, translate);
+            const workspaceUpgradeNavigationDetails = accountingIntegrationData?.workspaceUpgradeNavigationDetails;
+            if (workspaceUpgradeNavigationDetails && !isControlPolicy(policy)) {
+                Navigation.navigate(
+                    ROUTES.WORKSPACE_UPGRADE.getRoute(policyID, workspaceUpgradeNavigationDetails.integrationAlias, workspaceUpgradeNavigationDetails.backToAfterWorkspaceUpgradeRoute),
+                );
+                return;
+            }
+            setActiveIntegration({
+                ...newActiveIntegration,
+                key: Math.random(),
+            });
+        },
+        [policy, policyID, translate],
+    );
+
+    const closeConfirmationModal = () => {
+        setActiveIntegration((prev) => {
+            if (prev) {
+                return {
+                    ...prev,
+                    shouldDisconnectIntegrationBeforeConnecting: false,
+                    integrationToDisconnect: undefined,
+                };
+            }
+            return undefined;
+        });
+    };
+
+    const accountingContext = useMemo(
+        () => ({
+            activeIntegration,
+            startIntegrationFlow,
+            integrationRefs,
+        }),
```

Comment:
bump for this ^

Reply:
I'll remove memoization 👍 Probably a leftover

Reply:
It's because of our linting rules. It's required to memorize objects which are passed as Context props 🤷 

`react/jsx-no-constructed-context-values`